### PR TITLE
[WIP] Dialog setting

### PR DESCRIPTION
**Pull request by panaC**

### CODE REVIEW

File: src/renderer/library/components/settings/LanguageSettings.tsx [hunk 1]
```diff
@@ -29,43 +37,79 @@ interface IBaseProps extends TranslatorProps {
...
-                                    { AvailableLanguages[lang] }
-                                </label>
-                            </div>,
-                        )}
-                    </form>
-                </section>
-            </>
-        );
-    }
+interface AvailableLanguages {
```
Comment (panaC) [inline src/renderer/library/components/settings/LanguageSettings.tsx:57]: duplicate from https://github.com/edrlab/thorium-reader/blob/1f0decaa938b5cdf621da84804fdda193aed6bac/src/common/services/translator.ts#L159

File: src/renderer/library/components/settings/LanguageSettings.tsx [hunk 2]
```diff
@@ -29,43 +37,79 @@ interface IBaseProps extends TranslatorProps {
...
 // ReturnType<typeof mapDispatchToProps>
 // eslint-disable-next-line @typescript-eslint/no-empty-interface
-interface IProps extends IBaseProps, ReturnType<typeof mapStateToProps>, ReturnType<typeof mapDispatchToProps> {
-}
+// interface IProps extends IBaseProps, ReturnType<typeof mapStateToProps>, ReturnType<typeof mapDispatchToProps> {
+// }
+
+const SelectItem = React.forwardRef(({ children, ...props }: any, forwardedRef) => {
+    return (
+        <Select.Item {...props} id="itemInner" ref={forwardedRef}
```
Comment (panaC) [inline src/renderer/library/components/settings/LanguageSettings.tsx:45]: You don't need to forwardRef a sub-component at the top of the tree

File: src/renderer/library/components/settings/Settings.tsx [hunk 3]
```diff
@@ -36,10 +43,48 @@ class Settings extends React.Component<IProps, undefined> {
                 <LibraryLayout
                     title={__("header.settings")}
                 >
```
Comment (panaC) [inline src/renderer/library/components/settings/Settings.tsx:45]: This component is loaded from react-router https://github.com/edrlab/thorium-reader/blob/1f0decaa938b5cdf621da84804fdda193aed6bac/src/renderer/library/routing.ts#L62-L66

I think that the react-router route `/setting` should be disable (the setting component is no longer a page but a modal dialog now). And Setting Dialog must be move to the `common/components/dialog` folder (used in library and reader as well)

Comment (danielweck) [reply src/renderer/library/components/settings/Settings.tsx:45]: > I think that the react-router route `/setting` should be disable (the setting component is no longer a page but a modal dialog now).

Nested routes (i.e. relative `settings_modal` path inside another route definition) would work well declaratively, for example `/bookshelf` + nested `/bookshelf/settings_modal` ... but the settings modal popup dialog with Radix is really designed to be displayed anywhere (overlaid on top of an existing route component) where a "call site" button (trigger UI control) can be placed, so I agree that using the router to implement this is a waste of time and effort (unless we really want to use router functionality, for example to open the settings dialog in a specific tab by passing route parameters?)

File: src/renderer/library/components/settings/ReadingOptions/TextContent.tsx [hunk 4]
```diff
@@ -0,0 +1,130 @@
+import fontList, { FONT_ID_DEFAULT, FONT_ID_VOID } from "readium-desktop/utils/fontList";
+import * as stylesReader from "readium-desktop/renderer/assets/styles/reader-app.css";
+
+const textContent= () => {
+    const {__, readerConfig} = this.props;
```
Comment (panaC) [inline src/renderer/library/components/settings/ReadingOptions/TextContent.tsx:5]: - The first letter of the component must be in the upperCase.
- You cannot used `this.props` from a react functional component, `props` must be passed from function parameter and not from function binding `this` this keyword is used in class components instead.

**Comment by panaC** [comment]
replaced by #2039

**Pull request closed without merging**